### PR TITLE
Remove batch consumer

### PR DESCRIPTION
**Pull request by BeckaL**
This was an experimental feature introduced in 2018 that was never actually used.

Relevant evidence:
* [Feature introduced](https://github.com/alphagov/govuk_message_queue_consumer/commit/c0d761e8f1f08dbf495360e24e7231f53b11f910) and noted in commit as 'proof of concept'
* [Unmerged PR from 2018](https://github.com/alphagov/search-api/pull/1141/files) to start using this feature in search api
* [The only references](https://github.com/search?q=org%3Aalphagov+BatchConsumer&type=code) in github to batch consumer are in this gem itself
* Additionally, we know the current implementation of the batch consumer would fail with ruby versions > 3 due to the changes to hashes and keyword arguments, and this gem is widely used, so presumably we would have run into issues with the ruby upgrades if this was used.

Note that this commit marks the change as unreleased - we will release this as a major version in a follow-up PR when the [change to swap CI to github actions](https://github.com/alphagov/govuk_message_queue_consumer/pull/72) is made in order to test the release process.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# Unreleased
```
Comment (kevindew) [inline CHANGELOG.md:1]: For what it's worth, merging changes like this is the "normal" approach we take to applying updates to gem's here. Bumping with a change should mostly be the exception, though I think that's slipped a bit in recent years.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
...
+
+- Remove unused batch consumer
```
Comment (kevindew) [inline CHANGELOG.md:3]: Should we add a "Breaking: " to this as a clue that we need to bump the major version.

Might want to drop the word unused here as we, the suppliers of the gem, just think it's not needed. We create a bit of a paradox if we describe it as unusued but remove it with a major change.

File: lib/govuk_message_queue_consumer.rb [hunk 3]
```diff
@@ -7,5 +7,4 @@
 require 'govuk_message_queue_consumer/message'
 require 'govuk_message_queue_consumer/message_consumer'
 require 'govuk_message_queue_consumer/consumer'
-require 'govuk_message_queue_consumer/batch_consumer'
```
Comment (kevindew) [inline lib/govuk_message_queue_consumer.rb:10]: Nice that the removal for this is quite clean 🎉

**Review by kevindew — COMMENTED** [review]
Just a changelog comment to resolve.

File: CHANGELOG.md (hunk 1 above)
Comment (BeckaL) [reply CHANGELOG.md:1]: Ah good to know 👍

File: CHANGELOG.md (hunk 2 above)
Comment (BeckaL) [reply CHANGELOG.md]: Yup all makes sense, have updated

**Pull request merged**